src/polyfem/assembler/PressureAssembler.hpp

```diff
@@ -0,0 +1,191 @@
+#pragma once
+
+#include <polyfem/assembler/Assembler.hpp>
+#include <polyfem/mesh/Obstacle.hpp>
+
+#include <polyfem/assembler/Problem.hpp>
+#include <polyfem/assembler/MatParams.hpp>
+#include <polyfem/mesh/LocalBoundary.hpp>
+
+namespace polyfem
+{
+	namespace assembler
+	{
+		class ThermodynamicProcess
+		{
+		public:
+			ThermodynamicProcess() {}
+			virtual ~ThermodynamicProcess(){};
+			virtual double pressure(const double start_pressure, const double v0, const double v1) const { return 0; }
+			virtual double first_derivative(const double start_pressure, const double v0, const double v1) const { return 0; }
+			virtual double second_derivative(const double start_pressure, const double v0, const double v1) const { return 0; }
+			virtual double energy(const double start_pressure, const double v0, const double v1) const { return 0; }
+
+		protected:
+			const double atmospheric = 101.3e3;
+		};
+
+		class IsothermalProcess : public ThermodynamicProcess
+		{
+		public:
+			virtual double pressure(const double start_pressure, const double v0, const double v1) const override
+			{
+				return ((start_pressure + atmospheric) * v0) / v1 - atmospheric;
+			}
+
+			virtual double first_derivative(const double start_pressure, const double v0, const double v1) const override
+			{
+				return -((start_pressure + atmospheric) * v0) / (v1 * v1);
+			}
+
+			virtual double second_derivative(const double start_pressure, const double v0, const double v1) const override
+			{
+				return 2 * ((start_pressure + atmospheric) * v0) / (v1 * v1 * v1);
+			}
+
+			virtual double energy(const double start_pressure, const double v0, const double v1) const override { return (start_pressure + atmospheric) * v0 * std::log(v1) - atmospheric * v1; }
+		};
+
+		class AdiabaticProcess : public ThermodynamicProcess
+		{
+		public:
+			virtual double pressure(const double start_pressure, const double v0, const double v1) const override
+			{
+				return ((start_pressure + atmospheric) * std::pow(v0 / v1, gamma_)) - atmospheric;
+			}
+
+			virtual double first_derivative(const double start_pressure, const double v0, const double v1) const override
+			{
+				return -gamma_ * (start_pressure + atmospheric) * std::pow(v0 / v1, gamma_) / v1;
+			}
+
+			virtual double second_derivative(const double start_pressure, const double v0, const double v1) const override
+			{
+				return gamma_ * (1 + gamma_) * (start_pressure + atmospheric) * std::pow(v0 / v1, gamma_) / v1 / v1;
+			}
+
+			virtual double energy(const double start_pressure, const double v0, const double v1) const override { return (start_pressure + atmospheric) * std::pow(v0, gamma_) * (std::pow(v1, 1. - gamma_)) / (1. - gamma_) - atmospheric * v1; }
+
+		private:
+			const double gamma_ = 1.4;
```

Comment:
user defined?

Reply:
Again, this is pretty much constant for normal air. Unless you plan on doing some very special things with specific single gases, I think this should remain hidden.